presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/PlanBuilder.java

```diff
@@ -0,0 +1,106 @@
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule.test;
+
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.sql.ExpressionUtils;
+import com.facebook.presto.sql.parser.SqlParser;
+import com.facebook.presto.sql.planner.PlanNodeIdAllocator;
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.plan.Assignments;
+import com.facebook.presto.sql.planner.plan.FilterNode;
+import com.facebook.presto.sql.planner.plan.LimitNode;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.facebook.presto.sql.planner.plan.ProjectNode;
+import com.facebook.presto.sql.planner.plan.ValuesNode;
+import com.facebook.presto.sql.tree.Expression;
+import com.google.common.collect.ImmutableList;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.lang.String.format;
+
+public class PlanBuilder
+{
+    private final PlanNodeIdAllocator idAllocator;
+    private final Map<Symbol, Type> symbols = new HashMap<>();
+
+    public PlanBuilder(PlanNodeIdAllocator idAllocator)
+    {
+        this.idAllocator = idAllocator;
+    }
+
+    public ValuesNode values(Symbol... columns)
+    {
+        return new ValuesNode(
+                idAllocator.getNextId(),
+                Stream.of(columns)
+                        .collect(Collectors.toList()),
+                ImmutableList.of());
+    }
+
+    public ValuesNode values(List<Symbol> columns, List<List<Expression>> rows)
+    {
+        return new ValuesNode(idAllocator.getNextId(), columns, rows);
+    }
+
+    public LimitNode limit(long limit, PlanNode source)
+    {
+        return new LimitNode(idAllocator.getNextId(), source, limit, false);
+    }
+
+    public ProjectNode project(Assignments assignments, PlanNode source)
+    {
+        return new ProjectNode(idAllocator.getNextId(), source, assignments);
+    }
+
+    public FilterNode filter(Expression predicate, PlanNode source)
+    {
+        return new FilterNode(idAllocator.getNextId(), source, predicate);
+    }
+
+    public Symbol symbol(String name, Type type)
+    {
+        Symbol symbol = new Symbol(name);
+
+        Type old = symbols.get(symbol);
+        if (old != null && !old.equals(type)) {
+            throw new IllegalArgumentException(format("Symbol '%s' already registered with type '%s'", name, old));
+        }
+
+        return symbol;
+    }
+
+    public static Expression expression(String sql)
+    {
+        return ExpressionUtils.rewriteQualifiedNamesToSymbolReferences(new SqlParser().createExpression(sql));
+    }
+
+    public static List<Expression> expressions(String... expressions)
+    {
+        return Stream.of(expressions)
+                .map(PlanBuilder::expression)
+                .collect(Collectors.toList());
```

Comment:
I thought that there is a convention to use `Immutable*` collections wherever it is possible